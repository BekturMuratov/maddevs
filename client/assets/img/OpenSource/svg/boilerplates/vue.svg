<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>