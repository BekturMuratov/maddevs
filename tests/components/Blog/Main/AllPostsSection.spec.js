import { render } from '@testing-library/vue'
import AllPostsSection from '@/components/Blog/Main/AllPostsSection'
import Vuex from 'vuex'
import { createLocalVue } from '@vue/test-utils'
import allPosts from '../../../__mocks__/allPosts'
import * as homeContent from '../../../__mocks__/homePageContent'

const localVue = createLocalVue()
localVue.use(Vuex)

const recipeMock = jest.fn()
recipeMock.mockReturnValue('homePageContent')

describe('AllPostsSection component', () => {
  it('is a Vue instance', () => {
    const { container } = render(AllPostsSection, {
      localVue,
      store: {
        getters: {
          filteredPosts: () => [{}],
          allPosts: () => [],
          postsCategory: jest.fn(),
          postsPage: jest.fn(),
          homePageContent: () => homeContent,
        },
      },
    })

    expect(container).toMatchSnapshot()
  })

  it('should correct render posts', () => {
    const { container } = render(AllPostsSection, {
      localVue,
      store: {
        getters: {
          filteredPosts: () => allPosts,
          allPosts: () => allPosts,
          postsCategory: jest.fn(),
          postsPage: jest.fn(),
          homePageContent: () => homeContent,
        },
      },
    })

    expect(container).toMatchSnapshot()
    // console.log(screen.getAllByTestId('test-single-post'))
  })
})
