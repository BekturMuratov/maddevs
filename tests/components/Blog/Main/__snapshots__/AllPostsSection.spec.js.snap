// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllPostsSection component is a Vue instance 1`] = `
<div>
  <!---->
</div>
`;

exports[`AllPostsSection component should correct render posts 1`] = `
<div>
  <div
    class="filtered-posts"
  >
    <div
      class="container"
    >
      <div
        class="filter"
      >
        <div
          data-simplebar="init"
        >
          <div
            class="simplebar-wrapper"
          >
            <div
              class="simplebar-height-auto-observer-wrapper"
            >
              <div
                class="simplebar-height-auto-observer"
              />
            </div>
             
            <div
              class="simplebar-mask"
            >
              <div
                class="simplebar-offset"
                style="right: 0px; bottom: 0px;"
              >
                <div
                  class="simplebar-content-wrapper"
                  style="height: auto; overflow-x: hidden; overflow-y: hidden;"
                >
                  <div
                    class="simplebar-content"
                  >
                    <ul
                      class="filter-list"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="simplebar-placeholder"
              style="width: 0px; height: 0px;"
            />
          </div>
           
          <div
            class="simplebar-track simplebar-horizontal"
            style="visibility: hidden;"
          >
            <div
              class="simplebar-scrollbar"
              style="width: 0px; display: none;"
            />
          </div>
           
          <div
            class="simplebar-track simplebar-vertical"
            style="visibility: hidden;"
          >
            <div
              class="simplebar-scrollbar"
              style="height: 0px; display: none;"
            />
          </div>
        </div>
      </div>
       
      <div
        class="filtered-posts__wrapper"
      />
       
      <!---->
    </div>
  </div>
</div>
`;
